docs/source/examples/checkpointer.rst

```diff
@@ -0,0 +1,451 @@
+  across the same number of files.
+
+One big advantage of being "state-dict invariant" is that you should be able to use
+fine-tuned checkpoints from TorchTune with any post-training tool (quantization, eval, inference)
+which supports the source format, without any code changes OR conversion scripts. This is one of the
+ways in which TorchTune interoperates with the surrounding ecosystem.
+
+To be "state-dict invariant", the ``load_checkpoint`` and
+``save_checkpoint`` methods make use of the weight convertors available
+`here <https://github.com/pytorch/torchtune/blob/main/torchtune/models/convert_weights.py>`_.
+
+|
+
+Handling different Checkpoint Formats
+-------------------------------------
+
+TorchTune supports three different
+`checkpointers <https://github.com/pytorch/torchtune/blob/main/torchtune/utils/_checkpointing/_checkpointer.py>`_,
+each of which supports a different checkpoint format.
+
+
+**HFCheckpointer**
+
+This checkpointer reads and writes checkpoints in a format which is compatible with the transformers
+framework from Hugging Face. As mentioned above, this is the most popular format within the Hugging Face
+Model Hub and is the default format in every TorchTune config.
+
+For this checkpointer to work correctly, we assume that checkpoint_dir contains the necessary checkpoint
+and json files. The easiest way to make sure everything works correctly is to use the following flow:
+
+- Download the model from the HF repo using tune download. By default, this will ignore the "safetensors"
+  files.
+
+    |
+
+    .. code-block:: bash
+
+        tune download meta-llama/Llama-2-7b-hf
+        --output-dir <checkpoint_dir>
+        --hf-token <hf-token>
+
+- Use ``output_dir`` above as the ``checkpoint_dir`` for the checkpointer.
+
+|
+
+The following snippet explains how the HFCheckpointer is setup in TorchTune config files.
+
+.. code-block:: yaml
+
+    checkpointer:
+
+        # checkpointer to use
+        _component_: torchtune.utils.FullModelHFCheckpointer
+
+        # directory with the checkpoint files
+        # this should match the output_dir above
+        checkpoint_dir: <checkpoint_dir>
+
+        # checkpoint files. For the llama2-7b-hf model we have
+        # 2 .bin files. The checkpointer takes care of sorting
+        # by id and so the order here does not matter
+        checkpoint_files: [
+            pytorch_model-00001-of-00002.bin,
+            pytorch_model-00002-of-00002.bin,
+        ]
+
+        # if we're restarting a previous run, we need to specify
+        # the file with the checkpoint state. More on this in the
+        # next section
+        recipe_checkpoint: null
+
+        # dir for saving the output checkpoints. Usually set
+        # to be the same as checkpoint_dir
+        output_dir: <checkpoint_dir>
+
+        # model_type which specifies how to convert the state_dict
+        # into a format which TorchTune understands
+        model_type: LLAMA2
```

Comment:
model_type is pretty confusing to me and not sure if its going to be super clear for users. For example, what's the supported list of `model_type`? If I want to work with mistral models, how do I find the appropriate `model_type`? Do I ever need to define my own?